app/graphql/types/profile.rb

```diff
@@ -22,14 +22,14 @@ class Profile < Types::BaseObject
     def rules(args = {})
       selected_columns = args[:lookahead].selections.map(&:name) &
                          ::Rule.column_names.map(&:to_sym)
-      rules = object.rules.select(selected_columns)
-
+      selected_columns << :id
```

Comment:
We have to add the `id` here. Otherwise the queries are done with "rule_id = NULL". So for example, computing compliance is done by (`SELECT * in rule_results ...long rank query... WHERE host_id=whatever AND rule_id=NULL` which will always be empty.

Maybe currently we don't see any issue with this in production because results and compliance are cached, but we will see it at the point that the `compliance` is recomputed.